internal/tenancy/tenancytest/namespace_test.go

```diff
@@ -42,12 +90,72 @@ func TestReadNamespace_NotFound(t *testing.T) {
 	client := svctest.RunResourceServiceWithConfig(t, config, tenancy.RegisterTypes)
 	cl := rtest.NewClient(client)
 
-	res := rtest.Resource(pbtenancy.NamespaceType, "ns1").
-		WithData(t, validNamespace()).Build()
+	cases := []struct {
+		name     string
+		resource *pbresource.Resource
+		errMsg   string
+	}{
+		{
+			name: "namespace with the given name not present",
+			resource: rtest.Resource(pbtenancy.NamespaceType, "ns1").
+				WithData(t, validNamespace()).
+				Build(),
+			errMsg: "resource not found",
+		},
+	}
 
-	_, err := cl.Read(context.Background(), &pbresource.ReadRequest{Id: res.Id})
-	require.Error(t, err)
-	require.Equal(t, codes.NotFound.String(), status.Code(err).String())
+	for _, tc := range cases {
+		t.Run(tc.name, func(t *testing.T) {
+			_, err := cl.Read(context.Background(), &pbresource.ReadRequest{Id: tc.resource.Id})
+			require.Error(t, err)
+			require.Equal(t, codes.NotFound.String(), status.Code(err).String())
+			require.Contains(t, err.Error(), tc.errMsg)
+		})
+	}
+}
+
+func TestReadNamespace_InvalidArgument(t *testing.T) {
```

Comment:
I think you can skip these tests if the generalized validation in the read endpoint (and it's test) already cover them.

Reply:
Okay, removed these tests.
Created [ENT PR](https://github.com/hashicorp/consul-enterprise/pull/7505) for a test case related to the wrong partition.